File: code/backend/app/presentation/schemas/sessao.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.domain.entities.sessao import Avaliacao, Sessao, StatusSessao


class SessaoCreateRequest(BaseModel):
    solicitante_id: str
    descricao: str = Field(min_length=1, max_length=2000)


class SessaoStatusUpdateRequest(BaseModel):
    status: StatusSessao
    ouvinte_id: Optional[str] = None


class SessaoResponse(BaseModel):
    id: str
    solicitante_id: str
    ouvinte_id: Optional[str] = None
    descricao: str
    status: StatusSessao
    avaliacao: Optional[Avaliacao] = None
    created_at: datetime
    updated_at: datetime

    @classmethod
    def from_entity(cls, entity: Sessao) -> "SessaoResponse":
        return cls(
            id=str(entity.id),
            solicitante_id=str(entity.solicitante_id),
            ouvinte_id=str(entity.ouvinte_id) if entity.ouvinte_id else None,
            descricao=entity.descricao,
            status=entity.status,
            avaliacao=entity.avaliacao,
            created_at=entity.created_at,
            updated_at=entity.updated_at,
        )

File: code/app_cliente/lib/models/sessao.dart
import 'avaliacao.dart';
import 'status_sessao.dart';

/// Espelha o `SessaoResponse` do backend.
class Sessao {
  final String id;
  final String solicitanteId;
  final String? ouvinteId;
  final String descricao;
  final StatusSessao status;
  final Avaliacao? avaliacao;
  final DateTime createdAt;
  final DateTime updatedAt;

  Sessao({
    required this.id,
    required this.solicitanteId,
    this.ouvinteId,
    required this.descricao,
    required this.status,
    this.avaliacao,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Sessao.fromJson(Map<String, dynamic> json) {
    return Sessao(
      id: json['id'] as String,
      solicitanteId: json['solicitante_id'] as String,
      ouvinteId: json['ouvinte_id'] as String?,
      descricao: json['descricao'] as String,
      status: StatusSessao.fromApi(json['status'] as String),
      avaliacao: json['avaliacao'] != null
          ? Avaliacao.fromJson(json['avaliacao'] as Map<String, dynamic>)
          : null,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }
}
